lib/hash.h

```diff
@@ -235,7 +235,9 @@ extern void *hash_release(struct hash *hash, void *data);
 /*
  * Iterate over the elements in a hash table.
  *
- * It is safe to delete items passed to the iteration function from the hash
+ * It is safe to delete 1 item passed to the iteration function from
+ * the hash table during each call of the passed in function.
```

Comment:
Not accurate, if you delete the item that comes after the one you're iterating on, then on the next iteration you'll still use the now-deleted hash bucket because we already stored that bucket as the next one to use

Reply:
If you delete more than 1 item at a time, it is not guaranteed to be safe.  It may be safe but you don't know till you crash, especially given vagaries of how items are placed in buckets.   Therefore I believe this statement is accurate.  

If you are saying I should draw the subtle distinction about how and when this may or not be safe I don't think it really matters as that if you delete more than one item per iteration you will eventually have a crash in FRR.

Reply:
fixed to better wording after converstaoin w/ @qlyoung